### DISCUSSION TITLE
terminal app support

### DISCUSSION THREAD

**Discussion by deng232** [Aylur/ags#464] [Ideas]
Weather  an app needs to be run inside a terminal can be known by using ```this._app.get_boolean("Terminal")```. The Exec ( name which used in terminal) can access by ```this.app.get_commandline()```; and Utilits should also support to access env variables 
```
export const getenv = GLib.getenv

export const get_environ = GLib.get_environ().map((str: string) => str.split("="))
    .reduce((acc: { [x: string]: string; }, c: string[]) => { acc[c[0]] = c[1]; return acc }, {} as Record<string, string>)
```

**Comment by kotontrion** [comment]
GLib does support the proposed xdg-terminal-exec spec. So if you have that installed, you can use it to define your preferred terminal. Calling `launch()`on the Application object will then correctly open it in that terminal. If xdg-teminal-exec is not installed GLib will fall back to it's internal hardcoded list of supported terminals.

**Reply by deng232** [reply]
Wait, does the current release support terminal app or not?

**Reply by kotontrion** [reply]
Yes, I can start a terminal apps like vim from my applauncher, which then opens vim in a new terminal window.

AGS does not handle that itself, it is handled by glib. Glib only supports a few terminal emulators. If you have none of the supported terminals installed, glib can't launch it. So you have to either install a supported terminal emulator, or install and configure xdg-terminal-exec.

**Reply by deng232** [reply]
it doesn't work on foot though, even with xdg-terminal-exec setup. gonna look for it

**Reply by kotontrion** [reply]
I had it working with foot before switching to wezterm. 

`~/.config/xdg-terminals.list` should contain a list .desktop files of your preferred terminals. eg like this:
```
org.wezfurlong.wezterm.desktop
org.codeberg.dnkl.foot.desktop
```

will first try wezterm, if it is not installed it tries foot

**Reply by deng232** [reply]
I got it fixed. I didn't use xdg-terminals.list. just setup xdg-terminals-exec to use foot and change the $TERM system variable

**Reply by deng232** [reply]
maybe add those things to document though
